portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -163,12 +163,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       long id = entity.getKey().getId();
       String name = (String) entity.getProperty(NAME);
       String email = (String) entity.getProperty(EMAIL);
+      String imageUrl = (String) entity.getProperty(IMAGE_URL);
       String commentText = (String) entity.getProperty(COMMENT_TEXT);
       long timeStamp = (long) entity.getProperty(TIME_STAMP);
       
       // Creates new Comment for JSON accessibility.
-      Comment newComment = Comment.create(id, name, email, commentText, timeStamp);
-      comments.add(newComment);
+      Builder commentBuilder = Comment.builder(id, name, email, commentText, timeStamp);
+      if (imageUrl.isEmpty()) {
+        comments.add(commentBuilder.build());
+        continue;
+      }
+      comments.add(commentBuilder.setImageUrl(Optional.of(imageUrl)).build());
```

Comment:
I think in this case it would be a little cleaner to do:

if (!imageUrl.isEmpty()) {
  commentBuilder.setImageUrl(Optional.of(imageUrl));
}

comments.add(commentBuilder.build());

Reply:
agreed.